addons/web_editor/controllers/main.py

```diff
@@ -526,6 +526,8 @@ def modify_image(self, attachment, res_model=None, res_id=None, name=None, data=
             fields['res_id'] = res_id
         if name:
             fields['name'] = name
+        if request.env.user.has_group('website.group_website_publisher'):
```

Comment:
Can't we use the already existing context key `binary_field_real_user`?

Reply:
Removed the `attachments_mime_force_xml_image`.

Reply:
cc @odony (pinged below)

@bso-odoo While using a context key seems to actually be a security breach, as mentioned in my [other comment](https://github.com/odoo/odoo/pull/92540#discussion_r1036243940), here I was speaking about `binary_field_real_user` which can't really be altered from the client/js side to add it yourself as you would have to add as value a `res.users` python record which seems impossible from a rpc/client call.

At least, that's what I am figuring reading the commit that introduced this context key: [1be50fdeafcd](https://github.com/odoo/odoo/commit/1be50fdeafcd2b94f7b2f47d6e5bc4b6ebaceadd#diff-ab3dadc37163820f8e863edb1dd216f8b0e7ccf19cf4e2052577a7bae7ddd7e8R257). I didn't find anything more about it in the related PR discussion.

(Note that later it was improved to raise an explicit error: [4ccf918c15](https://github.com/odoo/odoo/commit/4ccf918c1596ac0ee104774d98d014708f03e901))

So, this seems to be safe to use here, exactly as we already do below in this file: [/addons/web_editor/controllers/main.py#L665](https://github.com/odoo/odoo/blame/e5f735473bf86416fa9192b486ba3f2280a70cd8/addons/web_editor/controllers/main.py#L665)
```py
            # Need to bypass security check to write image with mimetype image/svg+xml
            # ok because svgs come from whitelisted origin
            context = {'binary_field_real_user': request.env['res.users'].sudo().browse([SUPERUSER_ID])}
            attachment = request.env['ir.attachment'].sudo().with_context(context).create({  # ...
```

So I was suggesting to only pass this context key if you are part of the `group_website_publisher` group and to not alter any more code, especially letting the `base` code untouched.
It seems unneeded to add another way of bypassing that check.
I pushed a temp commit on top of your PR (basically becoming a one line fix where this context is passed.) so we can further discuss this. It reverts your change and propose the fix I am talking about here.

@odony May I have your input here? I am missing something about that `binary_field_real_user` context key?

Reply:
@rdeodoo Here is a proposal that does not use `sudo` nor `binary_field_real_user`. It explicitly whitelists the xml images for `website.group_website_publisher` that ran through the `modify_image` route.